include/pybind11/cast.h

```diff
@@ -139,6 +153,95 @@ inline PyThreadState *get_thread_state_unchecked() {
 #endif
 }
 
+PYBIND11_NOINLINE inline bool try_implicit_conversion(const bool already_loaded[], const std::vector<PyObject *> &py_args) {
```

Comment:
Wouldn't it make more sense to use a ``PyObject *`` here, which stores a tuple (instead of std::vector<PyObject *>).  ``PyTuple_SETITEM`` generates just as much object code as std::vector operations, and then the resulting list of arguments is already in the right format to be used by the function call.